people-and-planet-ai/weather-forecasting/serving/data.py

```diff
@@ -0,0 +1,232 @@
+    """
+    window_start = (date - WINDOW).isoformat()
+    window_end = date.isoformat()
+    return (
+        ee.ImageCollection("NASA/GPM_L3/IMERG_V06")
+        .filterDate(window_start, window_end)
+        .select("precipitationCal")
+        .sort("system:time_start", False)
+        .mosaic()
+        .unmask(0)
+        .float()
+    )
+
+
+def get_gpm_sequence(dates: list[datetime]) -> ee.Image:
+    """Gets a Global Precipitation Measurement sequence for the selected dates.
+
+    Args:
+        dates: List of dates to get images from.
+
+    Returns: An Earth Engine image.
+    """
+    images = [get_gpm(date) for date in dates]
+    return ee.ImageCollection(images).toBands()
+
+
+def get_goes16(date: datetime) -> ee.Image:
+    """Gets a GOES 16 image for the selected date.
+
+    For more information:
+        https://developers.google.com/earth-engine/datasets/catalog/NOAA_GOES_16_MCMIPF
+
+    Args:
+        date: Date to take a snapshot from.
+
+    Returns: An Earth Engine image.
+    """
+    window_start = (date - WINDOW).isoformat()
+    window_end = date.isoformat()
+    return (
+        ee.ImageCollection("NOAA/GOES/16/MCMIPF")
+        .filterDate(window_start, window_end)
+        .select("CMI_C.*")
+        .sort("system:time_start", False)
+        .mosaic()
+        .unmask(0)
+        .float()
+    )
+
+
+def get_goes16_sequence(dates: list[datetime]) -> ee.Image:
+    """Gets a GOES 16 sequence for the selected dates.
+
+    Args:
+        dates: List of dates to get images from.
+
+    Returns: An Earth Engine image.
+    """
+    images = [get_goes16(date) for date in dates]
+    return ee.ImageCollection(images).toBands()
+
+
+def get_elevation() -> ee.Image:
+    """Gets a digital elevation map.
+
+    For more information:
+        https://developers.google.com/earth-engine/datasets/catalog/MERIT_DEM_v1_0_3
+
+    Returns: An Earth Engine image.
+    """
+    return ee.Image("MERIT/DEM/v1_0_3").rename("elevation").unmask(0).float()
+
+
+def get_inputs_image(date: datetime) -> ee.Image:
+    """Gets an Earth Engine image with all the inputs for the model.
+
+    Args:
+        date: Date to take a snapshot from.
+
+    Returns: An Earth Engine image.
+    """
+    dates = [date + timedelta(hours=h) for h in INPUT_HOUR_DELTAS]
+    precipitation = get_gpm_sequence(dates)
+    cloud_and_moisture = get_goes16_sequence(dates)
+    elevation = get_elevation()
+    return ee.Image([precipitation, cloud_and_moisture, elevation])
+
+
+def get_labels_image(date: datetime) -> ee.Image:
+    """Gets an Earth Engine image with the labels to train the model.
+
+    Args:
+        date: Date to take a snapshot from.
+
+    Returns: An Earth Engine image.
+    """
+    dates = [date + timedelta(hours=h) for h in OUTPUT_HOUR_DELTAS]
+    return get_gpm_sequence(dates)
+
+
+def get_inputs_patch(date: datetime, point: tuple, patch_size: int) -> np.ndarray:
+    """Gets the patch of pixels for the inputs.
+
+    args:
```

Comment:
NIT: Does this need to have initial caps?

Reply:
Good catch! Typo.